api/services/v1alpha1/servicemesh_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/opendatahub-io/opendatahub-operator/v2/api/common"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ServiceMeshServiceName = "servicemesh"
+	// ServiceMeshInstanceName the name of the ServiceMesh instance singleton.
+	// value should match whats set in the XValidation below
+	ServiceMeshInstanceName = "default-servicemesh"
+	ServiceMeshKind         = "ServiceMesh"
+)
+
+// Check that the component implements common.PlatformObject.
+var _ common.PlatformObject = (*ServiceMesh)(nil)
+
+type ServiceMeshSpec struct {
```

Comment:
nit: not for this PR but as follow up we could remove  `api/infrastructure/v1/servicemesh_types.go` since it is identical to this one.